src/main/java/frc/robot/subsystems/climber/Climber.java

```diff
@@ -0,0 +1,117 @@
+package frc.robot.subsystems.climber;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.can.TalonFX;
+import edu.wpi.first.wpilibj.Solenoid;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.Ports;
+import frc.robot.subsystems.UnitModel;
+
+public class Climber extends SubsystemBase {
+    private final TalonFX master = new TalonFX(Ports.Climber.MASTER);
+    private final TalonFX slave = new TalonFX(Ports.Climber.SLAVE);
+    private final UnitModel unitModel = new UnitModel(Constants.Climber.TICKS_PER_METER);
+    private final Solenoid stopper = new Solenoid(Ports.Climber.STOPPER);
+    private final Solenoid gearboxPiston = new Solenoid(Ports.Climber.GEARBOX_PISTON);
+
+    public Climber() {
+        slave.follow(master);
+
+        master.setInverted(Ports.Climber.MASTER_INVERTED);
+        slave.setInverted(Ports.Climber.SLAVE_INVERTED);
+
+        master.setSensorPhase(Ports.Climber.MASTER_SENSOR_PHASE_INVERTED);
+        slave.setSensorPhase(Ports.Climber.SLAVE_SENSOR_PHASE_INVERTED);
+
+        master.config_kP(0, Constants.Climber.kP, Constants.TALON_TIMEOUT);
+        master.config_kI(0, Constants.Climber.kI, Constants.TALON_TIMEOUT);
+        master.config_kD(0, Constants.Climber.kD, Constants.TALON_TIMEOUT);
+
+
+    }
+
+    /**
+     * Get the climber's elevation relative to the ground.
+     *
+     * @return the climber's height [m].
+     */
+    public double getHeight() {
+        return unitModel.toUnits(master.getSelectedSensorPosition());
+    }
+
+    /**
+     * Climb height meters up with the robot.
+     *
+     * @param height requested height to climb [m].
+     */
+    public void setHeight(double height) {
+        master.set(ControlMode.MotionMagic, unitModel.toTicks(height));

```

Comment:
For motion magic you also need to configure acceleration and cruise velocity in the constructor